persistence-api/src/main/java/io/stargate/db/Persistence.java

```diff
@@ -44,6 +44,8 @@
 
   ClientState newClientState(String name);
 
+  AuthenticatedUser<?> newAuthenticatedUser(String name);
```

Comment:
Opening this API does not sound right to me. Since we validate credentials (token) via `AuthenticationService` service and by-pass the negotiator from `Persistence` in this case, would it be preferable to have a local implementation of `AuthenticatedUser` (in the `cql` module) for this use case?